src/lib/Export.hs

```diff
@@ -237,9 +227,10 @@ instance Show (ExportType n) where
     ScalarType       sbt       -> showExportSBT sbt
     where
       showShape shape = "[" <> (intercalate "," $ fmap showDim shape) <> "]"
-      showDim size = case size of
-        Left  name -> pprint name
-        Right lit  -> show lit
+      showDim size = case ixTypeType size of
+        Var  name -> pprint name
+        Fin n -> show n
+        ty -> pprint ty
```

Comment:
I think only `Fin n` should be allowed here, and the thing we want to print is exactly `n`. This is why shapes used to be lists of `Either (AtomName n) Int`, since those are the valid right-hand-sides of a `Fin`